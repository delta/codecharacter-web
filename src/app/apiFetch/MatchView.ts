/* tslint:disable:no-console*/
import { API_BASE_URL } from '../../config/config';
import { jsonResponseWrapper, setRequestHeaders } from './utils';

export const getMatches = (pageNo: number, pageSize: number) => {
  return fetch(`${API_BASE_URL}user/match/${pageNo}/${pageSize}`, {
    credentials: 'include',
    headers: setRequestHeaders(),
    method: 'GET',
  })
    .then((response) => {
      console.log('match response');
      console.log(response);

      return response.json();
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const getTopMatches = (pageNo: number, pageSize: number) => {
  return fetch(`${API_BASE_URL}match/top/${pageNo}/${pageSize}`, {
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'GET',
  })
    .then((response) => {
      console.log('top response');
      console.log(response);

      return response.json();
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const getGameLogs = (gameId: number) => {
  return fetch(`${API_BASE_URL}game/log/${gameId}`, {
    credentials: 'include',
    headers: setRequestHeaders(),
    method: 'GET',
  })
    .then((response) => {
      console.log(response);
      return jsonResponseWrapper(response);
    })
    .then((data) => data)
    .catch((error) => {
      console.error(error);
    });
};
