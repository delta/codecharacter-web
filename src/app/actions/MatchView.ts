import { Match } from 'app/types/MatchView';
import { action } from 'typesafe-actions';

export namespace MatchActions {
  export enum Type {
    GET_MATCHES = 'GET_MATCHES',
    GET_GAME_LOGS = 'GET_GAME_LOGS',
    GET_TOP_MATCHES = 'GET_TOP_MATCHES',
    VIEW_MATCH = 'VIEW_MATCH',
    UPDATE_LOADING_STATUS = 'UPDATE_LOADING_STATUS',
    UPDATE_ERROR = 'UPDATE_ERROR',
    UPDATE_MATCHES = 'UPDATE_MATCHES',
    UPDATE_TOP_MATCHES = 'UPDATE_TOP_MATCHES',
    RESET_MATCH_VIEW = 'RESET_MATCH_VIEW',
  }

  export const getMatches = (pageNo: number, pageSize: number) =>
    action(Type.GET_MATCHES, { pageNo, pageSize });

  export const getTopMatches = (pageNo: number, pageSize: number) =>
    action(Type.GET_TOP_MATCHES, { pageNo, pageSize });

  export const getGameLogs = (gameId: number) => action(Type.GET_GAME_LOGS, { gameId });

  export const updateLoadingStatus = (loadingStatus: boolean) => {
    return action(Type.UPDATE_LOADING_STATUS, { loadingStatus });
  };

  export const updateMatches = (matchesData: Match[]) => {
    return action(Type.UPDATE_MATCHES, { matchesData });
  };

  export const updateTopMatches = (topMatchesData: Match[]) => {
    return action(Type.UPDATE_TOP_MATCHES, { topMatchesData });
  };

  export const updateError = (err: string) => {
    return action(Type.UPDATE_ERROR, err);
  };

  export const resetMatchView = () => action(Type.RESET_MATCH_VIEW);
}
