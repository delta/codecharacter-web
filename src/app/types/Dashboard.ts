import { DashboardActions } from 'app/actions';
import { SidePanelTab } from 'app/reducers/Dashboard';
import { ActionType } from 'typesafe-actions';

const actions = {
  resetDashboardState: DashboardActions.resetDashboardState,
  setIsWelcomeModalOpen: DashboardActions.setIsWelcomeModalOpen,
  setSidePanelTab: DashboardActions.setSidePanelTab,
};

export enum SplitPaneState {
  EDITOR = 'EDITOR',
  BOTH = 'BOTH',
  RENDERER = 'RENDERER',
}

export interface State {
  editorWidthRatio: number;
  fixedLeftPaneWidth: number;
  rendererHeight: number;
  splitPaneState: SplitPaneState;
  windowWidth: number;
  isReactTourActive: boolean;
}

export interface StateProps {
  isLoggedIn: boolean;
  sidePanelOpen: boolean;
  isAuthenticationOpen: boolean;
  isUserProfileModalOpen: boolean;
  isWelcomeModalOpen: boolean;
  logFile: string;
  player1DebugLog: string;
  player2DebugLog: string;
}

export interface DispatchProps {
  setIsAuthenticationOpen: (isAuthenticationOpen: boolean) => void;
  closeWelcomeModal: () => void;
}

export type Props = StateProps & DispatchProps;

export type DashboardActions = Omit<typeof DashboardActions, 'Type'>;

export interface DashboardStoreState {
  sidePanelTab: SidePanelTab;
  isWelcomeModalOpen: boolean;
}

export type DashboardStoreAction = ActionType<typeof actions>;
