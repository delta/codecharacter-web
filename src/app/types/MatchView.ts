import { MatchActions } from 'app/actions';
import { ActionType } from 'typesafe-actions';

export enum MatchViewTabType {
  MY_MATCHES = 'MY_MATCHES',
  TOP_MATCHES = 'TOP_MATCHES',
}

export interface OwnState {
  activeMatchViewTab: MatchViewTabType;
  pageNo: number;
}

export interface Game {
  id: number;
  verdict: string;
  mapId: number;
  winType: string;
}

export interface Match {
  avatar1: string;
  avatar2: string;
  username1: string;
  score1: number;
  rating1?: number;
  username2: string;
  score2: number;
  rating2?: number;
  verdict: string;
  playedAt: string;
  match_mode: string;
  games: Game[];
}

export interface StateProps {
  matches: Match[];
  topMatches: Match[];
  loading: boolean;
  currentUsername: string;
}

export interface DispatchProps {
  getMatches: (pageNo: number, pageSize: number) => void;
  getTopMatches: (pageNo: number, pageSize: number) => void;
  getGameLogs: (gameId: number) => void;
}

export type Props = StateProps & DispatchProps;

export type State = OwnState;

export interface ElementOwnProps {
  match: Match;
  index: number;
  currentUserMatch: boolean;
  type: MatchViewTabType;
  getGameLogs: (gameId: number) => void;
}

export interface ElementState {
  isSelected: boolean;
}

export type ElementProps = ElementOwnProps;

const actions = {
  getMatches: MatchActions.getMatches,
  getTopMatches: MatchActions.getTopMatches,
  resetMatchView: MatchActions.resetMatchView,
  updateError: MatchActions.updateError,
  updateLoading: MatchActions.updateLoadingStatus,
  updateMatches: MatchActions.updateMatches,
  updateTopMatchs: MatchActions.updateTopMatches,
};

export interface MatchStoreState {
  loading: boolean;
  matches: Match[];
  topMatches: Match[];
}

export interface RecievedGame {
  createdAt: string;
  id: number;
  interestingness: number;
  mapId: number;
  matchId: number;
  points1: number;
  points2: number;
  status: string;
  verdict: string;
}

export interface RecievedMatch {
  avatar1: number;
  avatar2: number;
  games: RecievedGame[];
  matchMode: string;
  playedAt: string;
  score1: number;
  score2: number;
  username1: string;
  username2: string;
  verdict: string;
}

export type MatchStoreAction = ActionType<typeof actions>;
