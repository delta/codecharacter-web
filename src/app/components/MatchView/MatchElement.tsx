import { faShieldAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as styles from 'app/styles/MatchView.module.css';
import { Avatar } from 'app/types/Authentication/Register';
import * as MatchInterfaces from 'app/types/MatchView';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';

export class MatchElement extends React.Component<
  MatchInterfaces.ElementProps,
  MatchInterfaces.ElementState
> {
  public MAP_THUMBNAIL = [
    'islesofcodechar',
    'timeturner',
    'treacherousmangroves',
    'mysterymap',
    'mysterymap',
  ];

  public MAP_NAME = [
    'Isles of Codecharacter',
    'Time Turner',
    'Treacherous Mangroves',
    'The Fuhrer',
    'Crossroads of Destiny',
  ];

  public constructor(props: MatchInterfaces.ElementProps) {
    super(props);
    this.state = {
      isHoveredOver: false,
    };
  }

  public render() {
    const { match, getGameLogs, type } = this.props;
    const { isHoveredOver } = this.state;

    const isMyMatch = type === MatchInterfaces.MatchViewTabType.MY_MATCHES;
    let matchResult = 'TIE';

    if (match.verdict === '0') {
      matchResult = 'TIE';
    } else if (match.verdict === '1') {
      matchResult = 'WON';
    } else {
      matchResult = 'LOST';
    }

    return (
      <Col
        sm={12}
        style={{ borderRadius: 5 }}
        className={classnames('mb-1')}
        onMouseEnter={() => {
          this.setState({
            isHoveredOver: true,
          });
        }}
        onMouseLeave={() => {
          this.setState({
            isHoveredOver: false,
          });
        }}
      >
        <div
          className={classnames(
            'w-100',
            styles.matchElement,
            type === MatchInterfaces.MatchViewTabType.TOP_MATCHES ? styles.topMatchElement : '',
            {
              [`${styles.winBackground}`]: isMyMatch && matchResult === 'WON',
              [`${styles.loseBackground}`]: isMyMatch && matchResult === 'LOST',
              [`${styles.tieBackground}`]: isMyMatch && matchResult === 'TIE',
            },
          )}
        >
          <Row className="d-flex justify-content-around ">
            <Col sm={5} style={{ textAlign: 'center' }}>
              <p style={{ fontSize: 13, margin: 0 }}>{`${match.username1.substr(0, 15)}${
                match.username1.length > 15 ? '...' : ''
              }`}</p>
              {match.rating1 ? (
                <p style={{ fontSize: 16 }}>
                  <b>{`${match.rating1}`}</b>
                </p>
              ) : null}
            </Col>
            <Col sm={2} style={{ textAlign: 'center' }}>
              <FontAwesomeIcon icon={faShieldAlt} />
            </Col>
            <Col sm={5} style={{ textAlign: 'center' }}>
              <p style={{ fontSize: 13, margin: 0 }}>{`${match.username2.substr(0, 15)}${
                match.username2.length > 15 ? '...' : ''
              }`}</p>
              {match.rating2 ? (
                <p style={{ fontSize: 16 }}>
                  <b>{`${match.rating2}`}</b>
                </p>
              ) : null}
            </Col>
          </Row>
          <div className={classnames(' d-flex justify-content-center w-100')}>
            {
              // @ts-ignore
              <img width={50} height={50} src={Avatar[match.avatar1]} />
            }
            <span className="text-capitalize text-font-weight-bold  h3 mx-3">
              {match.score1} - {match.score2}
            </span>
            {
              // @ts-ignore
              <img width={50} height={50} src={Avatar[match.avatar2]} />
            }
          </div>
          {isMyMatch ? (
            <div className=" d-flex justify-content-center text-capitalize text-font-weight-bold ">
              {matchResult}
            </div>
          ) : (
            <div className="d-flex" style={{ padding: 10 }} />
          )}
          <div className={classnames(styles.body, 'd-flex justify-content-around ')}>
            {match.games.map((game, index) => {
              let gameVerdict = 'TIE';

              if (game.verdict === '0') {
                gameVerdict = 'TIE';
              } else if (game.verdict === '1') {
                gameVerdict = 'WON';
              } else if (game.verdict === '2') {
                gameVerdict = 'LOST';
              } else {
                gameVerdict = 'RUNTIME ERROR';
              }

              return (
                <span
                  key={index}
                  className={classnames(styles.game)}
                  onClick={() => getGameLogs(game.id)}
                  title={
                    isMyMatch
                      ? `${this.MAP_NAME[game.mapId - 1]} - ${gameVerdict} - ${game.winType}`
                      : `${this.MAP_NAME[game.mapId - 1]}`
                  }
                >
                  <img
                    src={`assets/img/maps/${this.MAP_THUMBNAIL[game.mapId - 1]}.png`}
                    width={30}
                    height={30}
                    style={{
                      border: `4px solid ${
                        game.verdict === '0'
                          ? '#957f00'
                          : game.verdict === '1'
                          ? '#047104'
                          : game.verdict === '2'
                          ? '#770000'
                          : 'black'
                      }`,
                      borderRadius: 3,
                      marginBottom: 5,
                    }}
                  />
                </span>
              );
            })}
          </div>
          {isHoveredOver && isMyMatch ? (
            <div
              className=" d-flex justify-content-start text-capitalize  my-1 mx-3 h6"
              style={{
                fontSize: '12px',
                margin: '5px',
              }}
            >
              {new Date(match.playedAt).toLocaleString()}
            </div>
          ) : null}
        </div>
      </Col>
    );
  }
}
