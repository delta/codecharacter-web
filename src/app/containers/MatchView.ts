import { RootState } from 'app/reducers';

import { MatchActions } from 'app/actions';
import { Match } from 'app/components/MatchView';
import * as MatchInterfaces from 'app/types/MatchView';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    currentUsername: rootState.user.username,
    loading: rootState.match.loading,
    matches: rootState.match.matches,
    topMatches: rootState.match.topMatches,
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    getGameLogs: (gameId: number) => dispatch(MatchActions.getGameLogs(gameId)),
    getMatches: (pageNo: number, pageSize: number) =>
      dispatch(MatchActions.getMatches(pageNo, pageSize)),
    getTopMatches: (pageNo: number, pageSize: number) =>
      dispatch(MatchActions.getTopMatches(pageNo, pageSize)),
  };
};

const matchContainer = connect<MatchInterfaces.StateProps, MatchInterfaces.DispatchProps, {}>(
  mapStateToProps,
  mapDispatchToProps,
)(Match);

export default matchContainer;
