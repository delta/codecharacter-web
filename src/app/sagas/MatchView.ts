/* tslint:disable:no-console*/
import { GameLogActions, MatchActions, NotificationActions } from 'app/actions';
import * as MatchFetch from 'app/apiFetch/MatchView';
import { checkAuthentication, mapMatchResponse } from 'app/sagas/utils';
import { resType } from 'app/types/sagas';
import { all, call, put, takeEvery } from 'redux-saga/effects';
import { ActionType } from 'typesafe-actions';
import { isArray } from 'util';

export function* getMatches(action: ActionType<typeof MatchActions.getMatches>) {
  try {
    yield put(MatchActions.updateMatches([]));
    yield put(MatchActions.updateLoadingStatus(true));
    const res = yield call(MatchFetch.getMatches, action.payload.pageNo, action.payload.pageSize);
    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (isArray(res)) {
      const matchData = mapMatchResponse(res);

      yield put(MatchActions.updateMatches(matchData));
    } else yield put(NotificationActions.error(res.message));
  } catch (err) {
    console.error('get matches err');
    console.error(err);
  }
}

export function* getTopMatches(action: ActionType<typeof MatchActions.getTopMatches>) {
  try {
    yield put(MatchActions.updateTopMatches([]));
    yield put(MatchActions.updateLoadingStatus(true));
    const res = yield call(
      MatchFetch.getTopMatches,
      action.payload.pageNo,
      action.payload.pageSize,
    );

    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (isArray(res)) {
      const matchData = mapMatchResponse(res);

      yield put(MatchActions.updateTopMatches(matchData));
    } else yield put(NotificationActions.error(res.message));
  } catch (err) {
    console.error('top mat err');
    console.error(err);
  }
}

export function* getGameLogs(action: ActionType<typeof MatchActions.getGameLogs>) {
  try {
    const res = yield call(MatchFetch.getGameLogs, action.payload.gameId);

    const isAuthenticated = yield checkAuthentication(res);
    if (isAuthenticated === false) return;

    if (res.type === resType.ERROR) {
      yield put(MatchActions.updateError(res.error));
    } else {
      yield put(GameLogActions.setHideDebugLog(true));
      yield put(GameLogActions.clearDisplayDebugLog());

      const logs = res.logs;
      const debugLog1 = logs.player1Log;
      const debugLog2 = logs.player2Log;
      const gameLog = logs.gameLog;
      const matchPlayerId = logs.matchPlayerId;

      yield put(GameLogActions.updateGameLog('', '', ''));
      yield put(GameLogActions.updateGameLog(debugLog1, debugLog2, gameLog));
      yield put(GameLogActions.updateMatchPlayerId(matchPlayerId));
    }
  } catch (err) {
    console.error(err);
  }
}

export function* matchSagas() {
  yield all([
    takeEvery(MatchActions.Type.GET_MATCHES, getMatches),
    takeEvery(MatchActions.Type.GET_TOP_MATCHES, getTopMatches),
    takeEvery(MatchActions.Type.GET_GAME_LOGS, getGameLogs),
  ]);
}
