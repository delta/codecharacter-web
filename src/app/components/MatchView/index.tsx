import { faCaretLeft, faCaretRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { MatchElement } from 'app/components/MatchView/MatchElement';
import * as styles from 'app/styles/MatchView.module.css';
import * as MatchInterfaces from 'app/types/MatchView';
import classnames from 'classnames';
import * as React from 'react';
import { Col, Grid, Row } from 'react-bootstrap';

export class Match extends React.Component<MatchInterfaces.Props, MatchInterfaces.State> {
  private static paginationSize = 15;
  constructor(props: MatchInterfaces.Props) {
    super(props);
    this.state = {
      activeMatchViewTab: MatchInterfaces.MatchViewTabType.MY_MATCHES,
      pageNo: 1,
    };
  }

  public componentDidMount() {
    this.props.getMatches(this.state.pageNo, Match.paginationSize);
    this.props.getTopMatches(1, Match.paginationSize);
  }

  public handlePageClick = (dir: number) => {
    const { pageNo } = this.state;
    if (
      (dir === -1 && pageNo + dir > 0) ||
      (dir === 1 && this.props.matches.length === Match.paginationSize)
    ) {
      this.setState({
        pageNo: pageNo + dir,
      });
      this.getMatches(pageNo + dir);
    }
  };

  public compare(match1: MatchInterfaces.Match, match2: MatchInterfaces.Match) {
    const time1 = new Date(match1.playedAt).getTime();
    const time2 = new Date(match2.playedAt).getTime();

    return time2 - time1;
  }

  public render() {
    const { activeMatchViewTab } = this.state;
    const { matches, topMatches, getGameLogs, currentUsername } = this.props;

    matches.sort(this.compare);
    topMatches.sort(this.compare);

    return (
      <Grid fluid={true} className={classnames(styles.MatchView)}>
        <Row className="justify-content-between py-2 px-3">
          <Col className="text-light font-weight-bold my-auto">MATCHES</Col>
        </Row>
        <Row>
          <div className="col d-flex justify-content-center mb-3">
            <button
              className={classnames(styles.customBtn, {
                [`${styles.buttonActive}`]:
                  activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES,
              })}
              onClick={() => {
                this.toggleNotificationTab(MatchInterfaces.MatchViewTabType.MY_MATCHES);
                this.props.getMatches(this.state.pageNo, Match.paginationSize);
              }}
            >
              {' '}
              My Matches{' '}
            </button>
          </div>
          <div className="col d-flex justify-content-center mb-3">
            <button
              className={classnames(styles.customBtn, styles.topMatchesButton, {
                [`${styles.buttonActive}`]:
                  activeMatchViewTab === MatchInterfaces.MatchViewTabType.TOP_MATCHES,
              })}
              onClick={() => {
                this.toggleNotificationTab(MatchInterfaces.MatchViewTabType.TOP_MATCHES);
                this.props.getTopMatches(1, Match.paginationSize);
              }}
            >
              {' '}
              Top Matches{' '}
            </button>
          </div>
        </Row>
        <Row className={classnames('mb-2', styles.matchViewWrap)}>
          {activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES ? (
            matches && matches.length ? (
              matches.map((match, index) =>
                match ? (
                  <MatchElement
                    match={match}
                    index={index - 1}
                    key={index - 1}
                    getGameLogs={getGameLogs}
                    currentUserMatch={match.username1 === currentUsername}
                    type={MatchInterfaces.MatchViewTabType.MY_MATCHES}
                  />
                ) : (
                  <div className="ml-5"> Nothing to show </div>
                ),
              )
            ) : (
              <div className="ml-5"> Nothing to show </div>
            )
          ) : topMatches && topMatches.length ? (
            topMatches.map((match, index) =>
              match ? (
                <MatchElement
                  match={match}
                  index={index - 1}
                  key={index - 1}
                  getGameLogs={getGameLogs}
                  currentUserMatch={false}
                  type={MatchInterfaces.MatchViewTabType.TOP_MATCHES}
                />
              ) : (
                <div className="ml-5"> Nothing to show </div>
              ),
            )
          ) : (
            <div className="ml-5"> Nothing to show </div>
          )}
        </Row>
        {activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES &&
        matches.length > 0 ? (
          <Row>
            <Col
              className="d-flex justify-content-center"
              style={{ width: '100vw', margin: '10px' }}
            >
              <div>
                <span
                  className={classnames('px-2 py-1 border border-white')}
                  style={{ borderRadius: '16px', cursor: 'pointer' }}
                  onClick={() => this.handlePageClick(-1)}
                >
                  <FontAwesomeIcon icon={faCaretLeft} /> <FontAwesomeIcon icon={faCaretLeft} />
                </span>
                <span
                  className={classnames('mx-3 px-3 py-1')}
                  style={{
                    backgroundColor: 'white',
                    borderRadius: '16px',
                    color: 'black',
                    fontFamily: 'Karla',
                    fontWeight: 'bold',
                  }}
                >
                  {this.state.pageNo}
                </span>
                <span
                  className={classnames('px-2 py-1 border border-white')}
                  onClick={() => this.handlePageClick(1)}
                  style={{ borderRadius: '16px', cursor: 'pointer' }}
                >
                  <FontAwesomeIcon icon={faCaretRight} /> <FontAwesomeIcon icon={faCaretRight} />
                </span>
              </div>
            </Col>
          </Row>
        ) : null}
      </Grid>
    );
  }

  private getMatches(page: number) {
    if (this.state.activeMatchViewTab === MatchInterfaces.MatchViewTabType.MY_MATCHES) {
      this.props.getMatches(page, Match.paginationSize);
    }
  }

  private toggleNotificationTab = (activeMatchViewTab: MatchInterfaces.MatchViewTabType) => {
    this.setState({
      activeMatchViewTab,
    });
  };
}
