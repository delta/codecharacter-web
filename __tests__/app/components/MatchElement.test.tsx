import { MatchElement } from 'app/components/MatchView/MatchElement';
import { MatchViewTabType } from 'app/types/MatchView';
import { mount } from 'enzyme';
import * as React from 'react';

describe('MatchElement Component', () => {
  const wrapper = mount(
    <MatchElement
      currentUserMatch={false}
      index={1}
      match={{
        avatar1: 'BABOON',
        avatar2: 'BEAR',
        games: [],
        match_mode: 'SELF',
        playedAt: 'date string',
        score1: 1,
        score2: 5,
        username1: 'user1',
        username2: 'user2',
        verdict: '1',
      }}
      getGameLogs={(gameId: number) => undefined}
      type={MatchViewTabType.TOP_MATCHES}
    />,
  );

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
