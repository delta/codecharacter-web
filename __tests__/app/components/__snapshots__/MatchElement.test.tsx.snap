// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchElement Component Should render 1`] = `
<MatchElement
  currentUserMatch={false}
  getGameLogs={[Function]}
  index={1}
  match={
    Object {
      "avatar1": "BABOON",
      "avatar2": "BEAR",
      "games": Array [],
      "playedAt": "date string",
      "score1": 1,
      "score2": 5,
      "username1": "user1",
      "username2": "user2",
      "verdict": "1",
    }
  }
  type="TOP_MATCHES"
>
  <Col
    bsClass="col"
    className="mb-1"
    componentClass="div"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    sm={12}
    style={
      Object {
        "borderRadius": 5,
      }
    }
  >
    <div
      className="mb-1 col-sm-12"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderRadius": 5,
        }
      }
    >
      <div
        className="w-100"
      >
        <Row
          bsClass="row"
          className="d-flex justify-content-around "
          componentClass="div"
        >
          <div
            className="d-flex justify-content-around  row"
          >
            <Col
              bsClass="col"
              componentClass="div"
              sm={5}
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <div
                className="col-sm-5"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <p
                  style={
                    Object {
                      "fontSize": 13,
                      "margin": 0,
                    }
                  }
                >
                  user1
                </p>
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              sm={2}
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <div
                className="col-sm-2"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f3ed",
                        "M496 128c0 221.282-135.934 344.645-221.539 380.308a48 48 0 0 1-36.923 0C130.495 463.713 16 326.487 16 128a48 48 0 0 1 29.539-44.308l192-80a48 48 0 0 1 36.923 0l192 80A48 48 0 0 1 496 128zM256 446.313l.066.034c93.735-46.689 172.497-156.308 175.817-307.729L256 65.333v380.98z",
                      ],
                      "iconName": "shield-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-shield-alt fa-w-16 "
                    data-icon="shield-alt"
                    data-prefix="fas"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M496 128c0 221.282-135.934 344.645-221.539 380.308a48 48 0 0 1-36.923 0C130.495 463.713 16 326.487 16 128a48 48 0 0 1 29.539-44.308l192-80a48 48 0 0 1 36.923 0l192 80A48 48 0 0 1 496 128zM256 446.313l.066.034c93.735-46.689 172.497-156.308 175.817-307.729L256 65.333v380.98z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              sm={5}
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <div
                className="col-sm-5"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <p
                  style={
                    Object {
                      "fontSize": 13,
                      "margin": 0,
                    }
                  }
                >
                  user2
                </p>
              </div>
            </Col>
          </div>
        </Row>
        <div
          className=" d-flex justify-content-center w-100"
        >
          <img
            height={50}
            width={50}
          />
          <span
            className="text-capitalize text-font-weight-bold  h3 mx-3"
          >
            1
             - 
            5
          </span>
          <img
            height={50}
            width={50}
          />
        </div>
        <div
          className="d-flex"
          style={
            Object {
              "padding": 10,
            }
          }
        />
        <div
          className="d-flex justify-content-around "
        />
      </div>
    </div>
  </Col>
</MatchElement>
`;
